catroid/src/main/java/org/catrobat/catroid/stage/PenActor.java

```diff
@@ -140,4 +149,21 @@ private float calculateScreenRatio() {
 				+ Math.pow(header.getVirtualScreenHeight(), 2));
 		return creatorDiagonalPixel / deviceDiagonalPixel;
 	}
+
+	private static final int DELAYED_DRAW_CYCLES = 2;
+
+	private class DelayedFrameBufferCreator {
+		int numDrawCycles = 0;
+
+		void createBufferForDraw() {
+			if (numDrawCycles < DELAYED_DRAW_CYCLES) {
+				numDrawCycles++;
+			} else if (buffer == null) {
```

Comment:
could you move the if() into the brackets {}? costed me one WTF, til I got it.
this is high level complaining

your whole construction took me a while, but now I know what your evil plan is here :)